core/src/components/modal/modal.tsx

```diff
@@ -719,6 +719,37 @@ export class Modal implements ComponentInterface, OverlayInterface {
     return this.currentBreakpoint;
   }
 
+  private moveToNextBreakpoint() {
+    const { breakpoints, currentBreakpoint } = this;
+    if (breakpoints && currentBreakpoint !== undefined) {
+      const allowedBreakpoints = breakpoints.filter((b) => b !== 0);
+      const currentBreakpointIndex = allowedBreakpoints.indexOf(currentBreakpoint);
+      const nextBreakpointIndex = (currentBreakpointIndex + 1) % allowedBreakpoints.length;
+      /**
+       * Sets the current breakpoint to the next available breakpoint.
+       * If the current breakpoint is the last breakpoint, we set the current
+       * breakpoint to the first non-zero breakpoint to avoid dismissing the sheet.
+       */
+      this.setCurrentBreakpoint(allowedBreakpoints[nextBreakpointIndex]);
+    }
+  }
+
+  private onHandleClick = () => {
+    this.moveToNextBreakpoint();
+  };
+
+  private onHandleKeyUp = (ev: KeyboardEvent) => {
+    if (ev.key === 'Enter') {
+      this.moveToNextBreakpoint();
+    }
+  };
+
+  private onHandleKeyDown = (ev: KeyboardEvent) => {
+    if (ev.key === 'Enter') {
+      ev.preventDefault();
+    }
+  };
```

Comment:
What does this do? Typically pressing Enter/Space also dispatches a click event, so `onHandleClick` should be sufficient.